packages/oauth-client/src/EarthdataLoginClient.ts

```diff
@@ -0,0 +1,180 @@
+   *   clientId: 'my-client-id',
+   *   clientPassword: 'my-client-password',
+   *   earthdataLoginUrl: 'https://earthdata.login.nasa.gov',
+   *   redirectUri: 'http://my-api.com'
+   * });
+   */
+
+  constructor(
+    params: {
+      clientId: string,
+      clientPassword: string,
+      earthdataLoginUrl: string,
+      redirectUri: string
+    }
+  ) {
+    if (!params.clientId) throw new TypeError('clientId is required');
+    if (!params.clientPassword) throw new TypeError('clientPassword is required');
+    if (!params.earthdataLoginUrl) throw new TypeError('earthdataLoginUrl is required');
+    if (!params.redirectUri) throw new TypeError('redirectUri is required');
+
+    super({
+      clientId: params.clientId,
+      clientPassword: params.clientPassword,
+      loginUrl: params.earthdataLoginUrl,
+      redirectUri: params.redirectUri,
+    });
+
+    this.clientId = params.clientId;
+    this.clientPassword = params.clientPassword;
+    validateUrl(params.earthdataLoginUrl);
+    this.earthdataLoginUrl = params.earthdataLoginUrl;
+    validateUrl(params.redirectUri);
+    this.redirectUri = params.redirectUri;
+  }
+
+  /**
+   * Query the Earthdata Login API for the UID associated with a token
+   *
+   * @param {Object} params
+   * @param {string} params.onBehalfOf - the Earthdata Login client id of the
+   *   app requesting the username
+   * @param {string} params.token - the Earthdata Login token
+   * @param {string} [params.xRequestId] - a string to help identify the request
+   *   in the Earthdata Login logs
+   * @returns {Promise<string>} the UID associated with the token
+   */
+  async getTokenUsername(params: {
+    onBehalfOf: string,
+    token: string,
+    xRequestId?: string
+  }) {
+    const { onBehalfOf, token, xRequestId } = params;
+
+    const headers = xRequestId ? { 'X-Request-Id': xRequestId } : undefined;
+
+    try {
+      const response = <VerifyTokenResponse>(await super.postRequest({
+        loginPath: 'oauth/tokens/user',
+        headers,
+        form: {
+          client_id: this.clientId,
+          on_behalf_of: onBehalfOf,
+          token,
+        },
+      }));
+
+      return response.body.uid;
+    } catch (error) {
+      if (error instanceof got.ParseError) {
+        throw new EarthdataLoginError(
+          'InvalidResponse',
+          'Response from Earthdata Login was not valid JSON'
+        );
+      }
+
+      if (isHttpForbiddenError(error)) {
+        throw httpErrorToEarthdataLoginError(error);
+      }
+
+      throw error;
+    }
+  }
+
+  /**
+   * Given an authorization code, request an access token and associated
+   * information from the EarthdataLogin service. This overrides the
+   * base class for better, EarthdataLogin-specific errors.
+   *
+   * See OAuthClient.getAccessToken(authorizationCode).
+   *
+   * @param {string} authorizationCode - an OAuth2 authorization code
+   * @returns {Promise<Object>} access token information
+   */
```

Comment:
Same comments as fore Cognito client.  This override method can be removed if we override methods to convert the error.

Reply:
Yup, good plan. Addressed in previous commits.